### PR TITLE
Unnotify detector

### PR DESCRIPTION
**Pull request by jeremyjams**

### CODE REVIEW

File: src/main/java/com/iexec/core/detector/replicate/UnnotifiedAbstractDetector.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+import java.util.List;
+import java.util.Optional;
+
+import static com.iexec.common.replicate.ReplicateStatus.WORKER_LOST;
+import static com.iexec.common.replicate.ReplicateStatus.getMissingStatuses;
+
+@Slf4j
+@Service
+public class UnnotifiedAbstractDetector {
```
Comment (Ugo) [inline src/main/java/com/iexec/core/detector/replicate/UnnotifiedAbstractDetector.java]: The class should be _abstract_.

File: src/main/java/com/iexec/core/detector/replicate/UnnotifiedAbstractDetector.java [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+                                                       ChainContributionStatus onchainCompleted) {
+        for (Task task : taskService.findByCurrentStatus(dectectWhenOffchainTaskStatuses)) {
+            for (Replicate replicate : replicatesService.getReplicates(task.getChainTaskId())) {
+                Optional<ReplicateStatus> lastRelevantStatus = replicate.getLastRelevantStatus();
+
+                if (!lastRelevantStatus.isPresent()) {
+                    continue;
+                }
+
+                boolean isReplicateStatusContributing = lastRelevantStatus.get().equals(offchainCompleting);
```
Comment (Ugo) [inline src/main/java/com/iexec/core/detector/replicate/UnnotifiedAbstractDetector.java]: isReplicateStatusCompleting

File: src/main/java/com/iexec/core/detector/replicate/UnnotifiedAbstractDetector.java [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+                                 ChainContributionStatus onchainCompleted) {
+        for (Task task : taskService.findByCurrentStatus(dectectWhenOffchainTaskStatuses)) {
+            for (Replicate replicate : replicatesService.getReplicates(task.getChainTaskId())) {
+                Optional<ReplicateStatus> lastRelevantStatus = replicate.getLastRelevantStatus();
+
+                if (!lastRelevantStatus.isPresent()) {
+                    continue;
+                }
+
+                boolean isNotOffChainContributed = !lastRelevantStatus.get().equals(offchainCompleted);//avoid eth node call if already contributed
```
Comment (Ugo) [inline src/main/java/com/iexec/core/detector/replicate/UnnotifiedAbstractDetector.java]: isNotOffChainCompleted

File: src/main/java/com/iexec/core/detector/replicate/UnnotifiedAbstractDetector.java (hunk 2 above)
Comment (jeremyjams) [reply src/main/java/com/iexec/core/detector/replicate/UnnotifiedAbstractDetector.java]: Yes!

File: src/main/java/com/iexec/core/detector/replicate/UnnotifiedAbstractDetector.java (hunk 3 above)
Comment (jeremyjams) [reply src/main/java/com/iexec/core/detector/replicate/UnnotifiedAbstractDetector.java]: Yes2!

File: src/main/java/com/iexec/core/detector/replicate/UnnotifiedAbstractDetector.java (hunk 1 above)
Comment (jeremyjams) [reply src/main/java/com/iexec/core/detector/replicate/UnnotifiedAbstractDetector.java]: Right, thanks :)

**Pull request merged**